### PR TITLE
Pull COMPANY profile from the datasource

### PR DESCRIPTION
**Pull request by dmitrypintusov**
Add a method for IEX which will populate COMPANY data either for all stocks (if no ticker given) or a particular stock (if ticker given). Here is API endpoint data:
https://iexcloud.io/docs/api/#company

### CODE REVIEW

File: datawell/iex.py [hunk 1]
```diff
@@ -31,6 +33,36 @@ def get_stocks(self):
...
             raise ex
 
+    def get_company_data(self, company_symbol=None):
+        """
+        method for IEX which will populate COMPANY data
+        either for all stocks (if no ticker given) or a particular stock (if ticker given).
+        :return: a list of companies or a particular company
+        """
+        try:
+            return self.get_all_companies() if company_symbol is None else self.get_company_by_symbol(company_symbol)
```
Comment (aprotsik) [inline datawell/iex.py]: Is there a certain plan in having 2 separate methods that are used here? I don't think they'll be re-used somewhere else

Comment (dmitrypintusov) [reply datawell/iex.py]: @aprotsik according to the description "Add a method for IEX which will populate COMPANY data either for all stocks (if no ticker given) or a particular stock (if ticker given)" - as I've got from this we need a method that will perform in a different way for ticker given/no ticker given. 

get_all_companies() method reuse the logic from get_company_by_symbol(symbol) and the idea behind this approach is to remove code duplication.

**Comment by dmitrypintusov** [comment]
Refactored the code, please review one more time.

File: datawell/iex.py [hunk 2]
```diff
@@ -31,6 +32,14 @@ def get_stocks(self):
             ex = app.AppException(e, message)
             raise ex
 
+    def get_companies(self):
+        for company_symbol in self.Symbols:
```
[the remark this answers is not in the document]
Comment (petelind) [reply datawell/iex.py]: We kind of assume __init__ will work as supposed, right, and Symbols will be populated. But what if it failed, and the Symbols are empty?

File: datawell/iex.py [hunk 3]
```diff
@@ -31,6 +32,14 @@ def get_stocks(self):
...
+            company_symbol['company_info'] = self.get_company_by_symbol(company_symbol['symbol'])
+
+    def get_company_by_symbol(self, company_symbol: str):
+        uri = app.BASE_API_URL + '/stock/{}/company'.format(company_symbol) + app.API_TOKEN
```
[the remark this answers is not in the document]
Comment (dmitrypintusov) [reply datawell/iex.py]: Fixed.

File: datawell/iex.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmitrypintusov) [reply datawell/iex.py]: Added check for empty self.Symbols and exception handling while iterating over it.

File: datawell/iex.py (hunk 1 above)
Comment (petelind) [reply datawell/iex.py]: @dmitrypintusov, in general, it makes perfect sense, but in this particular case I'd rather support @aprotsik - it took me quite a few minutes to understand the intent (maybe the absence of the docstrings is not helping?.. :))
I'd rather do it shorter and without the get_company_by_symbol.

Comment (dmitrypintusov) [reply datawell/iex.py]: Simplified to one method get_companies().

File: datawell/iex.py [hunk 4]
```diff
@@ -31,6 +32,17 @@ def get_stocks(self):
             ex = app.AppException(e, message)
             raise ex
 
+    def get_companies(self):
+        if self.Symbols:
```
Comment (nikolai-riabinin-epam) [inline datawell/iex.py]: if you want to skipp all block use 
`if bad_condition: return None`

File: datawell/iex.py [hunk 5]
```diff
@@ -31,6 +32,17 @@ def get_stocks(self):
...
+            try:
+                for company_symbol in self.Symbols:
+                    uri = app.BASE_API_URL + 'stock/{}/company'.format(company_symbol['symbol']) + app.API_TOKEN
+                    company_symbol['company_info'] = self.load_from_iex(uri)
+            except Exception as e:
```
Comment (nikolai-riabinin-epam) [inline datawell/iex.py]: I think now you have no exceptions here (NoneType error skipped before)

**Pull request merged**